python/whylogs/api/fugue/profiler.py

```diff
@@ -0,0 +1,101 @@
+        self,
+        partition,
+        cols,
+        dataset_timestamp: Optional[datetime] = None,
+        creation_timestamp: Optional[datetime] = None,
+        profile_field: str = DF_PROFILE_FIELD,
+    ):
+        now = datetime.utcnow()
+
+        self._dataset_timestamp = dataset_timestamp or now
+        self._creation_timestamp = creation_timestamp or now
+
+        self._partition = PartitionSpec(partition)
+        self._by = self._partition.partition_by
+        self._cols = cols
+        self._profile_field = profile_field
+        self._profile_schema = Schema([(profile_field, bytes)])
+
+    def to_col_profiles(self, df: pd.DataFrame) -> Iterable[Dict[str, Any]]:
+        res = why.log(df[self._cols] if self._cols is not None else df)
+        for col_name, col_profile in res.view().get_columns().items():
+            yield {COL_NAME_FIELD: col_name, COL_PROFILE_FIELD: col_profile.serialize()}
+
+    def merge_col_profiles(self, df: pd.DataFrame) -> pd.DataFrame:
+        merged_profile: ColumnProfileView = reduce(
+            lambda acc, x: acc.merge(x), df[COL_PROFILE_FIELD].apply(ColumnProfileView.deserialize)
+        )
+        return df.head(1).assign(**{COL_PROFILE_FIELD: merged_profile.serialize()})
+
+    def merge_to_view(self, col_profiles: List[Dict[str, Any]]) -> Iterable[Dict[str, Any]]:
+        profile_view = DatasetProfileView(
+            columns={row[COL_NAME_FIELD]: ColumnProfileView.from_bytes(row[COL_PROFILE_FIELD]) for row in col_profiles},
+            dataset_timestamp=self._dataset_timestamp,
+            creation_timestamp=self._creation_timestamp,
+        )
+        yield {self._profile_field: profile_view.serialize()}
+
+    def transform_no_logical_partition(self, df: Any, **kwargs: Any) -> DatasetProfileView:
+        dag = FugueWorkflow()
+        input_df = dag.load(df) if isinstance(df, str) else dag.df(df)
+        cols = input_df.partition(self._partition).transform(
+            self.to_col_profiles,
+            schema=[(COL_NAME_FIELD, str), (COL_PROFILE_FIELD, bytes)],
+        )
+        merged_cols = cols.partition_by(COL_NAME_FIELD).transform(self.merge_col_profiles, schema="*")
+        result = merged_cols.process(self.merge_to_view, schema=self._profile_schema)
+        result.yield_dataframe_as("result", as_local=True)
+        return DatasetProfileView.deserialize(dag.run(**kwargs)["result"].as_array()[0][0])
+
+    def profile_partition(self, df: pd.DataFrame) -> pd.DataFrame:
+        res = why.log(df[self._cols] if self._cols is not None else df).view().serialize()
+        return df.head(1)[self._by].assign(**{self._profile_field: res})
+
+    def transform_with_logical_partition(self, df: Any, **kwargs: Any) -> Any:
+        return transform(
+            df,
+            transformer(lambda pdf: pdf.schema.extract(self._by) + self._profile_schema)(self.profile_partition),
+            partition=self._partition,
+            **kwargs,
+        )
+
+
+def profile(
```

Comment:
I'd prefer a less generic name to make this clear that this is for Fugue. Especially as this is in the API. How about something like FugueDatasetProfile or FugueProfiler?

Reply:
The method would still need to be imported from whylogs.api.fugue which helps clarify this is in context of a fugue integration, so I think it is ok.